src/mixed-line-bar-chart/bar-series.tsx

```diff
@@ -126,30 +135,68 @@ export default function BarSeries<T extends ChartDataTypes>({
           [styles['series--dimmed']]: highlightedXValue !== null && !matchesX(highlightedXValue, series.data[i].x),
         });
 
-        return axis === 'x' ? (
-          <rect
-            key={`bar-${i}`}
-            fill={color}
-            x={x + widthOffset / 2}
-            y={y + heightOffset / 2}
-            width={width - widthOffset}
-            height={height - heightOffset}
-            rx={rx}
-            className={className}
-          />
-        ) : (
-          <rect
-            key={`bar-${i}`}
-            fill={color}
-            x={y - height + heightOffset / 2}
-            y={x + widthOffset / 2}
-            width={height - heightOffset}
-            height={width - widthOffset}
-            rx={rx}
-            className={className}
-          />
+        const rectPlacement =
+          axis === 'x'
+            ? {
+                x: x + widthOffset / 2,
+                y: y + heightOffset / 2,
+                width: width - widthOffset,
+                height: height - heightOffset,
+              }
+            : {
+                x: y - height + heightOffset / 2,
+                y: x + widthOffset / 2,
+                width: height - heightOffset,
+                height: width - widthOffset,
+              };
+
+        // Non-stacked bars have all corners rounded.
+        // Stacked bars only require rounded corners for the min/max segments.
+        const rxProps = !stackedBarOffsets ? { rx } : { clipPath: `url(#${clipPathId}-${i})` };
+
+        return (
+          <React.Fragment key={`bar-${i}`}>
+            {/* Render clip paths to provide rounded corners for the min/max stacked bars  */}
+            {stackedBarOffsets && (isMin || isMax) && (
+              <defs aria-hidden="true">
+                <clipPath id={`${clipPathId}-${i}`}>
+                  {isMin && (
+                    <rect
+                      {...(axis === 'x' ? stretchRect(rectPlacement, 'down') : stretchRect(rectPlacement, 'left'))}
+                      rx={rx}
+                    />
+                  )}
+                  {isMax && (
+                    <rect
+                      {...(axis === 'x' ? stretchRect(rectPlacement, 'up') : stretchRect(rectPlacement, 'right'))}
+                      rx={rx}
+                    />
+                  )}
+                </clipPath>
+              </defs>
+            )}
+
+            {/* Render the bar rectangle */}
+            <rect fill={color} {...rectPlacement} {...rxProps} className={className} />
+          </React.Fragment>
         );
       })}
     </g>
   );
 }
+
+function stretchRect(
+  rect: { x: number; y: number; height: number; width: number },
+  direction: 'left' | 'right' | 'up' | 'down'
+) {
+  switch (direction) {
+    case 'up':
+      return { ...rect, height: rect.height + 10 };
```

Comment:
Where are these 10 numbers came from? I do not see it in the original code

Reply:
That's a good question. I will add a code comment to explain that.

Reply:
See the comment now but still do not understand what is happening? Why is this 10 pixels and not 5 or 100?